test/testdata/sonataflow.org_v1alpha08_sonataflow-simpleops.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Red Hat, Inc. and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: sonataflow.org/v1alpha08
+kind: SonataFlow
+metadata:
+  name: simple
+  annotations:
+    sonataflow.org/description: Simple example on k8s!
+    sonataflow.org/version: 0.0.1
+spec:
+  podSpecTemplate:
+    flowContainer:
+      image: quay.io/kiegroup/sonataflow-minimal-example:latest
```

Comment:
To double check with my previous question, so here what we are doing is:

1) In the flow, we are defining the HelloWorld workflow.

2) then, by, an external media we have already prepared an image that has exactly the build of that workflow. In the tests we have done all this manually by doing docker build -t quay.io/kiegroup/sonataflow-minimal-example:latest /tmp/simpleworkflow/ etc. So, we know by construction that the workflow is the same, that we have quarkus project build that can start, etc.

3) So in the end, we have an image that was produced by the quay.io/kiegroup/kogito-swf-builder-nightly:latest, pretty much the same that would have been done if no image was configured.

Finally, all scenarios where the workflow defined in the "flow" is different than the "corresponding" workflow already packaged/build in the provided image are not good. why? because, the semantic of what you see in the "flow" is different from what will be executing.

Maybe I'm missing something, but I see this very risky, and other than a kind of optimization I don't see the use cases for this. ¿?

Reply:
> Maybe I'm missing something, but I see this very risky, and other than a kind of optimization I don't see the use cases for this. ¿?

@wmedvede I believe I replied to your questions in the other comment. As far from this one, we can't control the build process. Users might have really strict and particular requirements when building images such as hardening, security issues, company policies, and so on. Opening an API to enable custom builds in the operator can be quite challenging and it's not our core. The build system today is simple enough for really small/simple use cases, but we can't scale. For this reason, we explore the `PodSpec` for users to set their image built externally.

As I said, we will add an integrity check to the image as we progress avoiding this kind of disruption.